### ISSUE TITLE
Add keep-alive to the transport (Ping/Pong)

### ISSUE THREAD

**Issue by mbonneau** [voryx/Thruway#8]

**Comment by oanhnn** [comment]
@mbonneau 
Package `cboden/ratchet` is released version 0.3.3 (https://github.com/ratchetphp/Ratchet/tree/v0.3.3) and you PR (https://github.com/ratchetphp/Ratchet/pull/232) isn't merged.
This problem has been solved yet?

I used your solution in #65 
```
"repositories": [
        {
            "type": "vcs",
            "url": "https://github.com/mbonneau/Ratchet"
        }
    ],
    "require": {
        "php": ">=5.4",
        "react/socket": "0.4.*",
        "cboden/ratchet": "dev-pong as 0.3.2",
        "ratchet/pawl": "0.1.*",
        "psr/log": "~1"
    },
```
but when `composer install`, i have a problem:
```
Loading composer repositories with package information                                                                                             
Initializing PEAR repository http://pear.cakephp.org                                                                                               
Reading composer.json of cboden/ratchet (v0.3.2)    
Reading composer.json of cboden/ratche  (v0.3.1)
Reading composer.json of cboden/ratchet (v0.2.8)
Reading composer.json of cboden/ratchet   0.2.7)
Reading composer.json of cboden/ratchet (v0.2.5)
Reading composer.json of cboden/ratchet (v  2.4)
Reading composer.json of cboden/ratchet (v0.2.2)
Reading composer.json of cboden/ratchet (v0.  1)
Reading composer.json of cboden/ratchet (v0.1.5)
Reading composer.json of cboden/ratchet (v0.1.  
Reading composer.json of cboden/ratchet (v0.1.2)
Reading composer.json of cboden/ratchet (v0.1)    
Reading composer.json of cboden/ratchet (0.4-wip/comp-interf  
Reading composer.json of cboden/ratchet (master)                 
Reading composer.json of cboden/ratchet (rfc-split)
 Reading composer.json of cboden/ratchet (wamp-sync-bug)
Installing dependencies (including require-dev)             
  Your requirements could not be resolved to an installable set of packages.                                                                         
    Problem 1                                                                                                                                        
      - The requested package cboden/ratchet could not be found in any version, there may be a typo in the package name.                             
    Problem 2                                                                                                                                        
      - ratchet/pawl v0.1.2 requires cboden/ratchet 0.3.* -> no matching package found.                                                              
      - ratchet/pawl v0.1.1 requires cboden/ratchet 0.3.* -> no matching package found.                                                              
      - ratchet/pawl v0.1 requires cboden/ratchet 0.3.* -> no matching package found.                                                                
      - Installation request for ratchet/pawl 0.1.* -> satisfiable by ratchet/pawl[v0.1, v0.1.1, v0.1.2].                                            
  Potential causes:                                                                                                                                  
   - A typo in the package name                                                                                                                      
   - The package is not available in a stable-enough version according to your minimum-stability setting                                             
     see <https://groups.google.com/d/topic/composer-dev/_g3ASeIFlrc/discussion> for more details.                                                   
  Read <https://getcomposer.org/doc/articles/troubleshooting.md> for further common problems. 
```

**Comment by mbonneau** [comment]
Hi @oanhnn ,

As far as the problem you are having above, I inadvertently removed the pong branch from my repo recently. It is back now and I think it should work properly.

As far as future plans, I think ratchet is working on replacing the underlying websocket code with https://github.com/ratchetphp/RFC6455 .  This library will have the capabilities needed to allow the ping/pong in the transport to work.

**Comment by oanhnn** [comment]
Thank @mbonneau !!!
I will plan updating my application.

**Comment by ftrrtf** [comment]
Any news/plans about adding Ping/Pong functionality to Thruway?

**Comment by dschissler** [comment]
The Ratchet project appears to be near death.

**Comment by mbonneau** [comment]
@ftrrtf - Yes - I will put some effort into getting a workable solution to this soon. I have put some work into the RFC6455 project with @cboden, the author of Ratchet (which is still pretty active).

**Comment by cboden** [comment]
> The Ratchet project appears to be near death.

This is news to me. Does @dschissler know something I don't?

**Comment by dschissler** [comment]
@cboden I doubt that much but I've simply been watching the progress and there doesn't appear to be much of it.  I ran spent a whole bunch of time wiring everything together in my [Webird](https://github.com/perchlayer/webird) foundation and since then I've been building my proprietary layer on top of this project.  I think that within the next 12 months that I'll be able to help out as I've already fixed several difficult bugs in other libraries, since I tend to him them every time.

**Comment by ftrrtf** [comment]
@mbonneau Can I help you with something?

**Comment by mbonneau** [comment]
This feature will be in the next Ratchet release.

**Comment by gsaqui** [comment]
@mbonneau is this blocked until Racket goes to 0.4 or is it possible to implement now?

**Comment by mbonneau** [comment]
@gsaqui - For the `RatchetTransportProvider` - yes. Internally we have been using the 0.4-dev (or 0.5) branch for about 2 years. We have found this to be pretty stable but does rely on development versions of dependencies.

If you do use the 0.5 branch you can enable keep alive on the `RatchetTransportProvider` using https://github.com/thruway/ratchet-transport/blob/437ceaf2fa6200306a5638a229a41a7d2e9bfebd/src/RatchetTransportProvider.php#L174

**Comment by gsaqui** [comment]
@mbonneau thank you for your help.

**Comment by davidwdan** [comment]
This has been resolved in v0.5.0
